core/Azure.Mcp.Core/src/Areas/Server/Commands/ServiceStartCommand.cs

```diff
@@ -734,4 +739,91 @@ private static WebApplication UseHttpsRedirectionIfEnabled(WebApplication app)
+    /// traces to user-configured Application Insights instance only when the necessary environment variables are set. This also honors 
+    /// the AZURE_MCP_COLLECT_TELEMETRY environment variable to allow users to disable telemetry collection if desired. Note that this is 
+    /// in addition to the telemetry configured in <see cref="OpenTelemetryExtensions"/>.
+    /// </remarks>
+    private static TracerProvider? ConfigureSelfHostingTelemetry(ServiceStartOptions options)
+    {
+#if RELEASE
+        if (options.Transport != TransportTypes.Http)
+        {
+            return null;
+        }
+
+        string? collectTelemetry = Environment.GetEnvironmentVariable("AZURE_MCP_COLLECT_TELEMETRY");
+        bool isTelemetryEnabled = string.IsNullOrWhiteSpace(collectTelemetry) ||
+            (bool.TryParse(collectTelemetry, out bool shouldCollectTelemetry) && shouldCollectTelemetry);
+
+        string? connectionString = Environment.GetEnvironmentVariable("APPLICATIONINSIGHTS_CONNECTION_STRING");
+        if (!isTelemetryEnabled || string.IsNullOrWhiteSpace(connectionString))
+        {
+            return null;
+        }
+
+        return Sdk.CreateTracerProviderBuilder()
+            // captures incoming HTTP requests
+            .AddAspNetCoreInstrumentation()
+            // captures outgoing HTTP requests with filtering
+            .AddHttpClientInstrumentation(o => o.FilterHttpRequestMessage = ShouldInstrumentHttpRequest)
+            .AddAzureMonitorTraceExporter(exporterOptions => exporterOptions.ConnectionString = connectionString)
+            .Build();
+#else
+        return null;
+#endif
+    }
+
+    /// <summary>
+    /// Determines whether an HTTP request should be instrumented for telemetry collection.
+    /// </summary>
+    /// <param name="request">The HTTP request message to evaluate.</param>
+    /// <returns>
+    /// <c>true</c> if the request should be instrumented; otherwise, <c>false</c>.
+    /// </returns>
+    /// <remarks>
+    /// This method filters out specific requests to prevent telemetry issues:
+    /// <list type="bullet">
+    /// <item><description>Application Insights ingestion endpoints (to avoid telemetry loops)</description></item>
+    /// <item><description>Requests where the parent span is from Azure SDK (to avoid duplicate spans)</description></item>
+    /// </list>
+    /// </remarks>
+    private static bool ShouldInstrumentHttpRequest(HttpRequestMessage request)
+    {
+        // Exclude Application Insights ingestion requests to skip requests that are made to AppInsights when sending telemetry.
+        // See related issue - https://github.com/Azure/azure-sdk-for-net/issues/45366#issuecomment-2278511391
+        if (request.RequestUri?.AbsoluteUri.Contains("applicationinsights.azure.com/v2.1/track", StringComparison.Ordinal) == true)
+        {
+            return false;
+        }
+
+        // Azure SDKs create their own client span before calling the service using HttpClient.
```

Comment:
Please link to the Azure Monitor SDK code that this was copied from. It'll be good for whoever comes next to be aware that this was not a novel decision.

Reply:
Call out and link to the code in `UseAzureMonitor` and say that we're copying this without a team consensus on if this is desired for our project. Leave room for someone else to decide to change this behavior, add a configuration, or describe the parity that will come from `UseAzureMonitor` that would let someone delete this redundant check.